### PR TITLE
Fix reference code and add NonceGen test vectors

### PR DESCRIPTION
**Pull request by jonasnick**

### CODE REVIEW

File: bip-musig2/reference.py [hunk 1]
```diff
@@ -352,6 +355,29 @@ def test_key_agg_vectors():
     assert key_agg([X[0], X[0], X[0]], [], []) == expected[2]
     assert key_agg([X[0], X[0], X[1], X[1]], [], []) == expected[3]
 
+
+def test_nonce_gen_vectors():
+    def fill(i):
+        return (i).to_bytes(1,byteorder='big')*32
```
Comment (real-or-random) [inline bip-musig2/reference.py]:
```suggestion
        return i.to_bytes(1, byteorder='big') * 32
```

**Review by real-or-random — APPROVED** [review]
ACK 8ccd8ecf6d76781be67c026bbc15f8bdbc896797

**Comment by real-or-random** [comment]
Wait, this matches the expected outputs in https://github.com/ElementsProject/secp256k1-zkp/pull/192 only partly? E.g. `6285697E57E8578FCFDE6FE7FA4B6CA8E3A5AECAB4FB92F84B7F9DF8DDEF4DB8` is present in the -zkp PR but `F7827E966E36DB7A6098F73F9CD094C0DDD912CE1F24439D9D290E97846A0815` is not.

File: bip-musig2/reference.py (hunk 1 above)
Comment (jonasnick) [reply bip-musig2/reference.py]: I think the parantheses around `i` are required.

**Comment by jonasnick** [comment]
Sorry I forgot to push a fix to -zkp. Previously the hashed indices were not identical: (0,1) in secp-zkp, (1,2) here.

**Comment by real-or-random** [comment]
Ah, this explains the offset. :)

File: bip-musig2/reference.py (hunk 1 above)
Comment (real-or-random) [reply bip-musig2/reference.py]: Ok, may depend on the python version, it works for me without. Let's keep it. 

May still be good to add the spacing but that's really a micronit.

Comment (jonasnick) [reply bip-musig2/reference.py]: Oh I was wrong, parentheses are only required for literals. Fixed.

**Comment by robot-dreams** [comment]
ACK 9a48eec31224a38e02dbc4ba9c4371927fc69a75

By inspection:
- `nonce_hash` matches the spec update in #9 
- The test vectors here match the ones in https://github.com/ElementsProject/secp256k1-zkp/pull/192

**Comment by jonasnick** [comment]
Added commit to hash `i - 1` in NonceGen instead of `i`.

**Review by real-or-random — APPROVED** [review]
ACK f4216927a6daf796e36a53b6d2c8bc2782016d21

**Pull request merged**